railties/lib/rails/generators/actions/create_migration.rb

```diff
@@ -19,6 +19,11 @@ def identical?
           exists? && File.binread(existing_migration) == render
         end
 
+        def invoke!
+          invoked_file = super
+          File.exist?(@destination) ? invoked_file : relative_existing_migration
```

Comment:
Why is this needed?

Reply:
`CreateMigration` keeps an existing file when content is the same with an invoked file. In that case, we need to return an existing file name. If do not, `after_generate` callback will return a non-existing file name(invoked file name).